ops/ruby/devops-framework/tasks/count_monitoring_assets.rb

```diff
@@ -0,0 +1,66 @@
+require "google/cloud/monitoring"
+require 'logger'
+
+require_relative "./lib/service_account_manager"
+require_relative './lib/gcp_environment_visitor'
+
+CUSTOM_METRIC_FILTER = "metric.type = starts_with(\"custom.googleapis.com/\")"
+LOGS_BASED_METRIC_FILTER = "metric.type = starts_with(\"logging.googleapis.com/\")"
+USER_LOGS_BASED_METRIC_FILTER = "metric.type = starts_with(\"logging.googleapis.com/user/\")"
+
+logger = Logger.new(STDOUT)
+class MonitoringAssets
+  def initialize(envrionments_path_json, logger = Logger.new(STDOUT))
+    @visitor = GcpEnvironmentVisitor.new(envrionments_path_json, logger)
+    @logger = logger
+  end
+
+  # Demonstrate a simple usage of the AouEnvironmentVisitor with a few read-only operations
+  # on monitoring things in all the environments.
+  def inventory
+    @visitor.visit do |env|
+      counts  = {}
+      metric_client = Google::Cloud::Monitoring::Metric.new
+      metric_project_path = Google::Cloud::Monitoring::V3::MetricServiceClient.project_path(env.project_id)
```

Comment:
Not to throw a wrench into things, but did you look at whether Terraform could work for applying these? https://www.terraform.io/docs/providers/google/r/monitoring_group.html

(surface-level assessment as fine, there would be a lot of questions to answer here such as how we could run it and  how we'd do state tracking)

Reply:
RE log-based metrics:
This? https://www.terraform.io/docs/providers/google/r/logging_metric.html

RE Terraform:
While Terra is technically a prefix of Terraform, and there are usages of Terraform by Terra, it is an entirely independent and unrelated tool: https://www.terraform.io .

RE Feedback:
I'm sure the Stackdriver team also has an interest in receiving feedback/usage on their Terraform provider.